lib/src/executable.dart

```diff
@@ -58,6 +58,11 @@ Future<void> run(List<String> args) async {
     return;
   }
 
+  if (args.contains('sort_imports')) {
+    await handleSortImports(args);
+    return;
+  }
```

Comment:
Open to suggestions on how the FormatTool should kick off the import sorter.

`FormatTool#buildExecution` returns a `FormatExecution`, which is a declarative representation of `FormatTool` so that `FormatTool` can be easily tested. For this reason, I tried to keep import sorting out of the format tool itself, and chose to allow `dart_dev` to handle import sorting.

Reply:
I think my preference would be to handle this within the `FormatTool` rather than adding a hidden `sort_imports` command. I know we had to do that to support formatting from an IDE, but in this case I think we can accomplish it without.

I'm thinking we could keep FormatExecution the way it is in master so that it still just represents the `dart format`/`dartfmt` process, but then conditionally spawn the `ImportCleanerTool` within the `run()`.

```dart
  @override
  FutureOr<int> run([DevToolExecutionContext context]) async {
    context ??= DevToolExecutionContext();
    final formatExecution = buildExecution(
      context,
      configuredFormatterArgs: formatterArgs,
      defaultMode: defaultMode,
      exclude: exclude,
      formatter: formatter,
    );
    if (formatExecution.exitCode != null) {
      return formatExecution.exitCode;
    }
    var exitCode = await runProcessAndEnsureExit(formatExecution.process, log: _log);
    if (exitCode != 0) {
      return exitCode;
    }
    if (organizeImports) {
      exitCode = await ImportCleanerTool().run(context);
    }
    return exitCode;
  }
```

That way, we're only really adding one conditional branch of logic here and we can still unit test the `ImportCleanerTool` separately. If we want, we could add a couple of functional tests for formatting - one with and one without `organizeImports`.

Reply:
I guess we wouldn't be able to pass the args for the format tool directly through to the `ImportCleanerTool`. We could update the `context` that we pass through like so:

```dart
    if (organizeImports) {
      final importTool = ImportCleanerTool();
      context = context.update(argResults: importTool.argParser.parse([ ... ]);
      exitCode = await importTool.run(context);
    }
```

But that would require some refactoring to avoid duplicating the work to parse the formatter tool's args. Since we're not actually making `ImportCleanerTool` public and we're exposing this behavior via a flag on the `FormatterTool`, maybe we don't need the tool wrapper for this logic at all. We could instead just have this tool call into a utility that does the import organizing directly rather than constructing a tool and running it ourselves.

Maybe something like this? https://github.com/Workiva/dart_dev/compare/import_sort...import_sort_no_cleaner_tool?expand=1